ext/fiddle/pointer.c

```diff
@@ -370,6 +385,34 @@ rb_fiddle_ptr_free_get(VALUE self)
     return rb_fiddle_new_function(address, arg_types, ret_type);
 }
 
+/*
+ * call-seq: free! => nil
```

Comment:
How about renaming to `call_free` or something?
`free` (getter) and `free!` will confuse us.

Reply:
Yes... I was going for terseness, but I'm open to debating this.

Reply:
It's also a destructive operation, which fits the `!` pattern.

Reply:
`free!` is destructive but `free!` is not the destructive version of `free`.
The relation of them is different from the relation of `String#gsub` and `String#gsub!`.

Reply:
I feel like `free!` is best due to being short and looking destructive. I'm open to changing it if you're very certain or other people agree.

Reply:
If we have block style free feature (`Fiddle::Pointer.malloc(size, Fiddle::RUBY_FREE) {|pointer| ...}`?), we'll not call this method manually.
So I don't think we should use shorter name for this method with potential confusion. (Some users may use `pointer.free` wrongly to free a pointer but the pointer isn't freed. It causes a memory leak.)

Reply:
Ok I'll use `call_free`.

Reply:
Done.